### PR TITLE
Simplified regex used in check_modname_prefix and other improvements.

### PR DESCRIPTION
**Pull request by RobertZenz**
Simplified the regex used, added comments and changed the error message
to contain the correct mod name.

### CODE REVIEW

File: builtin/game/register.lua [hunk 1]
```diff
@@ -51,20 +51,24 @@ local forbidden_item_names = {
...
 		if name:sub(1, #expected_prefix) ~= expected_prefix then
 			error("Name " .. name .. " does not follow naming conventions: " ..
-				"\"modname:\" or \":\" prefix required")
+				"\"" .. expected_prefix .. "\" or \":\" prefix required")
 		end
+		
+		-- Enforce that the name only contains letters and underscores.
 		local subname = name:sub(#expected_prefix+1)
-		if subname:find("[^abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_]") then
+		if subname:find("[^a-zA-Z0-9_]") then
```
Comment (kilbith) [inline builtin/game/register.lua]: There's even simpler than that...
```Lua
subname:find("[%w_]")
```

Comment (kilbith) [inline builtin/game/register.lua]: Wouldn't be simpler to use `subname:find("[^%w_]")` ?

Comment (RobertZenz) [inline builtin/game/register.lua]: Right, I forgot that `%w` only includes this specific set, will change that.

Comment (RobertZenz) [inline builtin/game/register.lua]: Funny sidenote, Wikipedia says about `\w` that it contains the set `[a-zA-Z0-9_]`, but the Lua set of `%w` is only `[a-zA-Z0-9]`.

**Comment by kilbith** [comment]
This is so obvious that it should be merged directly. Don't let this rotting please. @est31 @BlockMen

**Comment by rubenwardy** [comment]
It's not a regex, it's a Lua pattern.

**Comment by est31** [comment]
:+1: if commit message uses present tense

**Comment by paramat** [comment]
:+1: With a better present-tense commit message.

**Comment by est31** [comment]
yeah, thats nrz problems.

**Comment by BlockMen** [comment]
> This is so obvious that it should be merged directly. Don't let this rotting please.

@kilbith ignoring obvious things is a speciality of MT's development...

Concerning this PR: :+1:

**Comment by est31** [comment]
merged with changed commit msg : 0cf15470fc16e1ebc13a7d8445cfd4278c3caf8b

**Pull request closed without merging**